include/zephyr/drivers/i3c/ccc.h

```diff
@@ -0,0 +1,1056 @@
+ * @param broadcast True if broadcast, false if direct.
+ */
+#define I3C_CCC_ENTAS(as, broadcast)		(((broadcast) ? 0x02U : 0x82U) + (as))
+
+/**
+ * Enter Activity State 0
+ *
+ * @param broadcast True if broadcast, false if direct.
+ */
+#define I3C_CCC_ENTAS0(broadcast)		I3C_CCC_ENTAS(0, broadcast)
+
+/**
+ * Enter Activity State 1
+ *
+ * @param broadcast True if broadcast, false if direct.
+ */
+#define I3C_CCC_ENTAS1(broadcast)		I3C_CCC_ENTAS(1, broadcast)
+
+/**
+ * Enter Activity State 2
+ *
+ * @param broadcast True if broadcast, false if direct.
+ */
+#define I3C_CCC_ENTAS2(broadcast)		I3C_CCC_ENTAS(2, broadcast)
+
+/**
+ * Enter Activity State 3
+ *
+ * @param broadcast True if broadcast, false if direct.
+ */
+#define I3C_CCC_ENTAS3(broadcast)		I3C_CCC_ENTAS(3, broadcast)
+
+/** Reset Dynamic Address Assignment (Broadcast) */
+#define I3C_CCC_RSTDAA				0x06U
+
+/** Enter Dynamic Address Assignment (Broadcast) */
+#define I3C_CCC_ENTDAA				0x07U
+
+/** Define List of Targets (Broadcast) */
+#define I3C_CCC_DEFTGTS				0x08U
+
+/**
+ * Set Max Write Length (Broadcast or Direct)
+ *
+ * @param broadcast True if broadcast, false if direct.
+ */
+#define I3C_CCC_SETMWL(broadcast)		((broadcast) ? 0x09U : 0x89U)
+
+/**
+ * Set Max Read Length (Broadcast or Direct)
+ *
+ * @param broadcast True if broadcast, false if direct.
+ */
+#define I3C_CCC_SETMRL(broadcast)		((broadcast) ? 0x0AU : 0x8AU)
+
+/** Enter Test Mode (Broadcast) */
+#define I3C_CCC_ENTTM				0x0BU
+
+/** Set Bus Context (Broadcast) */
+#define I3C_CCC_SETBUSCON			0x0CU
+
+/**
+ * Data Transfer Ending Procedure Control
+ *
+ * @param broadcast True if broadcast, false if direct.
+ */
+#define I3C_CCC_ENDXFER(broadcast)		((broadcast) ? 0x12U : 0x92U)
+
+/** Enter HDR Mode (HDR-DDR) (Broadcast) */
+#define I3C_CCC_ENTHDR(x)			(0x20U + (x))
+
+/** Enter HDR Mode 0 (HDR-DDR) (Broadcast) */
+#define I3C_CCC_ENTHDR0				0x20U
+
+/** Enter HDR Mode 1 (HDR-TSP) (Broadcast) */
+#define I3C_CCC_ENTHDR1				0x21U
+
+/** Enter HDR Mode 2 (HDR-TSL) (Broadcast) */
+#define I3C_CCC_ENTHDR2				0x22U
+
+/** Enter HDR Mode 3 (HDR-BT) (Broadcast) */
+#define I3C_CCC_ENTHDR3				0x23U
+
+/** Enter HDR Mode 4 (Broadcast) */
+#define I3C_CCC_ENTHDR4				0x24U
+
+/** Enter HDR Mode 5 (Broadcast) */
+#define I3C_CCC_ENTHDR5				0x25U
+
+/** Enter HDR Mode 6 (Broadcast) */
+#define I3C_CCC_ENTHDR6				0x26U
+
+/** Enter HDR Mode 7 (Broadcast) */
+#define I3C_CCC_ENTHDR7				0x27U
+
+/**
+ * Exchange Timing Information (Broadcast or Direct)
+ *
+ * @param broadcast True if broadcast, false if direct.
+ */
+#define I3C_CCC_SETXTIME(broadcast)		((broadcast) ? 0x28U : 0x98U)
+
+/** Set All Addresses to Static Addresses (Broadcast) */
+#define I3C_CCC_SETAASA				0x29U
+
+/**
+ * Target Reset Action
+ *
+ * @param broadcast True if broadcast, false if direct.
+ */
+#define I3C_CCC_RSTACT(broadcast)		((broadcast) ? 0x2AU : 0x9AU)
+
+/** Define List of Group Address (Broadcast) */
+#define I3C_CCC_DEFGRPA				0x2BU
+
```

Comment:
I'd be nice to have a define for I3C CCC Vendor commands
```
#define I3C_CCC_VENDOR(id, broadcast)	((id) + ((broadcast) ? 0x61 : 0xe0))
```

Reply:
Added